core/frontend/src/Viewport.ts

```diff
@@ -2494,6 +2494,51 @@ export abstract class Viewport implements IDisposable, TileUser {
     return this.target.readImageToCanvas();
   }
 
+  /** Waits for all tiles to load and render for this viewport.
+   * @returns A promise which will resolve when all tiles finish loading and are rendered for this viewport.
+   * @param options The options governing how this method behaves. Particularly, if the `isAddedToViewManager` property is false (defaults to true), this method will pump the tile request scheduler itself.
+   */
+  public async waitForTilesToLoad(options: {isAddedToViewManager: boolean} = {isAddedToViewManager: true}): Promise<void> {
+    let haveNewTiles = true;
+    while (haveNewTiles) {
```

Comment:
This function is doing a lot of redundant work if the viewport is participating in the render loop - recreating scene, re-reqeusting tiles, etc - can that be reduced?

Reply:
@pmconne Sounds like a good idea to reduce that - the parts needed were largely lifted from dpta, where it is not actively participating in the render loop.  I'm thinking the options flag set by dpta should maybe signify more than it currently is (which just makes sure dpta makes the function pump the tile request queue).  Let me know if you think that sounds like a bad approach; otherwise I'll work on trimming down the non-dpta code path and test it.